### PR TITLE
Add tests for new relation, and update old tests to work with new pgbouncer charm

### PR DESCRIPTION
**Pull request by WRFitch**
## Proposal
Add & update tests for updated pgbouncer charm

This PR fixes the tests that are blocking [PGBouncer PR#31](https://github.com/canonical/pgbouncer-k8s-operator/pull/31)

## Context
- This code has already been reviewed in the pgbouncer-k8s charm repo

### CODE REVIEW

File: tests/integration/relations/pgbouncer_provider/application-charm/src/charm.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        logger.info(event.params)
+
+        relation_id = event.params["relation-id"]
+        relation_name = event.params["relation-name"]
+        if relation_name == self.first_database.relation_name:
+            relation = self.first_database
+        elif relation_name == self.second_database.relation_name:
+            relation = self.second_database
+        elif relation_name == self.database_clusters.relation_name:
+            relation = self.database_clusters
```
Comment (marceloneppel) [inline tests/integration/relations/pgbouncer_provider/application-charm/src/charm.py:104]: Does `self.database_clusters` exist? I couldn't find it in this charm code.

File: tests/integration/relations/pgbouncer_provider/helpers.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    Args:
+        ops_test: The ops test framework instance
+        application_name: The name of the application
+        relation_name: name of the relation to get connection data from
+        key: key of data to be retrieved
+        relation_id: id of the relation to get connection data from
+
+    Returns:
+        the that that was requested or None
```
Comment (marceloneppel) [inline tests/integration/relations/pgbouncer_provider/helpers.py:74]:
```suggestion
        the data that was requested or None
```

**Review by marceloneppel — APPROVED** [review]
Super! It's amazing work, Will! Thanks!

File: tests/integration/relations/pgbouncer_provider/application-charm/src/charm.py (hunk 1 above)
Comment (WRFitch) [reply tests/integration/relations/pgbouncer_provider/application-charm/src/charm.py]: nope, forgot to remove this. Good catch!

**Review by marceloneppel — APPROVED** [review]

**Review by taurus-forever — APPROVED** [review]
Incredible!

**Pull request merged**